keps/sig-instrumentation/4827-component-statusz/README.md

```diff
@@ -206,25 +220,105 @@ rules:
 * **Minimum Compatibility Version**: The minimum Kubernetes API version with which the component is designed to work seamlessly
 * **Useful Links**: Relative URLs to other essential endpoints, such as /healthz, /livez, /readyz, and /metrics, for deeper insights into the component's health, readiness, and performance metrics
 
-#### Sample response
+#### Data format: JSON (v1)
+
+This is the default and preferred format for programmatic access.
+
+##### Request
+* Method: **GET**
+* Endpoint: **/statusz**
+* Header: `Accept: application/json;v=v1` (or no Accept header)
+* Body: empty
+
+##### Response Body
+
+The response is a JSON object with the following structure:
+
+```json
+{
+  "apiVersion": "v1",
+  "componentName": "<component-name>",
+  "startTime": "<start-time>",
+  "upTime": "<uptime-duration>",
+  "goVersion": "<go-version>",
+  "binaryVersion": "<binary-version>",
+  "emulationVersion": "<emulation-version>",
+  "minimumCompatibilityVersion": "<min-compatibility-version>",
+  "paths": [
+    "/configz",
+    "/healthz",
+    "/livez",
+    "/metrics",
+    "/readyz",
+    "/metrics/slis"
+  ]
+}
+```
+
+##### Sample Response
+
+```json
+{
+  "apiVersion": "v1",
+  "componentName": "kube-apiserver",
+  "startTime": "2024-09-06T06:19:51Z",
+  "upTime": "30s",
+  "goVersion": "go1.23.0",
+  "binaryVersion": "1.31.0-beta.0.981+c6be932655a03b-dirty",
+  "emulationVersion": "1.31.0-beta.0.981",
+  "minimumCompatibilityVersion": "1.30.0",
+  "paths": [
+    "/configz",
+    "/healthz",
+    "/livez",
+    "/metrics",
+    "/readyz",
+    "/metrics/slis"
+  ]
+}
+```
+
+#### Data format: text
+
+This format is available for human readability.
+
+##### Request
+* Method: **GET**
+* Endpoint: **/statusz**
+* Header: `Accept: text/plain`
+* Body: empty
+
+##### Sample response
 
 ```
-Started: Fri Sep  6 06:19:51 UTC 2024
-Up: 0 hr 00 min 30 sec
-Go version: go1.23.0
-Binary version: 1.31.0-beta.0.981&#43;c6be932655a03b-dirty
-Emulation version: 1.31.0-beta.0.981
-Minimum Compatibility version: 1.30.0
-
-List of useful endpoints
+Start Time: Fri Sep  6 06:19:51 UTC 2024
+Uptime: 0 hr 00 min 30 sec
+Go Version: go1.23.0
+Binary Version: 1.31.0-beta.0.981&#43;c6be932655a03b-dirty
+Emulation Version: 1.31.0-beta.0.981
+Minimum Compatibility Version: 1.30.0
+
+Available Paths
 --------------
-configz:/configz
-healthz:/healthz
-livez:/livez
-metrics:/metrics
-readyz:/readyz
-sli metrics:/metrics/slis
+configz: /configz
+healthz: /healthz
+livez: /livez
+metrics: /metrics
+readyz: /readyz
+slis: /metrics/slis
 ```
+### API Versioning and Deprecation Policy
+
+The versioned `/statusz` endpoint will follow a deprecation policy to ensure a predictable lifecycle for consumers as the API evolves.
+
+The deprecation policy for a given API version (e.g., `v1`) is as follows:
+- A deprecated API version must continue to be served for no less than **12 months or 3 releases** (whichever is longer) after the deprecation is announced.
```

Comment:
> A deprecated API version must continue to be served for no less than **12 months or 3 releases** (whichever is longer) after the deprecation is announced.

That doesn't sound right for stable versions ... once we advertise something as stable, people should be able to integrate with it and expect to ~never have to revisit their integration

Our policy for other GA APIs is that they will remain supported as long as Kubernetes is at 1.x (which I expect to be for the duration of the project)

https://kubernetes.io/docs/reference/using-api/deprecation-policy/

> GA API versions may be marked as deprecated, but must not be removed within a major version of Kubernetes